lib/ansible/module_utils/dimensiondata.py

```diff
@@ -0,0 +1,367 @@
+    networks = driver.ex_list_network_domains(location=location)
+    network = filter(lambda x: x.name == name, networks)
+    if len(network) > 0:
+        return network[0]
+    else:
+        return None
+
+
+# ---------------------------------------------
+# Get a network domain object by its name or id
+# ---------------------------------------------
+def get_network_domain(driver, locator, location):
+    if is_uuid(locator):
+        net_id = locator
+    else:
+        networks = driver.ex_list_network_domains(location=location)
+        found_networks = filter(lambda x: x.name == locator, networks)
+        if len(found_networks) > 0:
+            net_id = found_networks[0].id
+        else:
+            return False
+    try:
+        return driver.ex_get_network_domain(net_id)
+    except DimensionDataAPIException:
+        return False
+
+
+# ---------------------------------------------
+# Get a VLAN object by its name or id
+# ---------------------------------------------
+def get_vlan(driver, locator, location, network_domain):
+    if is_uuid(locator):
+        vlan_id = locator
+    else:
+        vlans = driver.ex_list_vlans(location=location,
+                                     network_domain=network_domain)
+        found_vlans = filter(lambda x: x.name == locator, vlans)
+        if len(found_vlans) > 0:
+            vlan_id = found_vlans[0].id
+        else:
+            return False
+    try:
+        return driver.ex_get_vlan(vlan_id)
+    except:
+        return False
+
+
+# ----------------------------------------
+# Get a locations MCP version
+# ----------------------------------------
+def get_mcp_version(driver, location):
+    # Get location to determine if MCP 1.0 or 2.0
+    location = driver.ex_get_location_by_id(location)
+    if 'MCP 2.0' in location.name:
+        return '2.0'
+    return '1.0'
+
+
+# ---------------------
+# Test if valid v4 UUID
+# ---------------------
+def is_uuid(u, version=4):
+    try:
+        uuid_obj = UUID(u, version=version)
+        return str(uuid_obj) == u
+    except:
+        return False
+
+
+# --------------------------------------------
+# Expand public IP block to show all addresses
+# --------------------------------------------
+def expand_ip_block(block):
+    addresses = []
+    ip_r = block.base_ip.split('.')
+    last_quad = int(ip_r[3])
+    address_root = "%s.%s.%s." % (ip_r[0], ip_r[1], ip_r[2])
+    for i in range(int(block.size)):
+        addresses.append(address_root + str(last_quad + i))
+    return addresses
+
+
+# ---------------------------
+# Get public IP block details
+# ---------------------------
+def get_public_ip_block(module, driver, network_domain, block_id=False,
+                        base_ip=False):
+    # Block ID given, try to use it.
+    if block_id is not 'False':
+        try:
+            block = driver.ex_get_public_ip_block(block_id)
+        except DimensionDataAPIException:
```

Comment:
a) was just note, not required to pass review
b) if libcloud already requries 2.6 we can make an exception in the check as we do for other cloud modules.